### ISSUE TITLE
Doesn't understand the attrs `default` and `converter` arguments correctly

### ISSUE THREAD

**Issue by tristanlatr** [python/mypy#10998]
**Bug Report**

Mypy is complaining about incompatible assignment when it should not, because the default value assigned in the `attr.ib()` call will be passed to the converter (and be converted). But it seems that mypy do no currently understand that.

`attrs` docs about order of execution: https://www.attrs.org/en/stable/init.html#order-of-execution

**To Reproduce**

Code example

```python
from pathlib import Path
import attr
@attr.s
class Options:
    path: Path = attr.ib(default='.', converter=Path)
```

**Expected Behavior**

It should not trigger errors. 

**Actual Behavior**

Triggers error: `6: error: Incompatible types in assignment (expression has type "str", variable has type "Path")`

**Your Environment**

- Mypy version used: mypy 0.910
- Mypy command-line flags: Nothing special
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.8
- Operating system and version: MacOS catalina

**Comment by NeilGirdhar** [comment]
This seems to be totally broken:
```
from attrs import define, field

@define
class C:
    x: tuple[int] = field(converter=tuple)

c = C((1, 2))  # Argument 1 to "C" has incompatible type "Tuple[int, int]"; expected "Iterable[_T_co]"
c = C([1, 2])  # List item 0 has incompatible type "int"; expected "_T_co"; List item 1 has incompatible type "int"; expected "_T_co"
```

**Comment by thejcannon** [comment]
This is now a part of the typing spec: https://github.com/python/typing-council/issues/33
